### PR TITLE
Extend identifier documentation

### PR DESCRIPTION
**Pull request by malteneuss**
Part of #904

### CODE REVIEW

File: lib/Hakyll/Core/Identifier.hs [hunk 1]
```diff
@@ -32,6 +35,67 @@ import           GHC.Exts            (IsString, fromString)
...
+@
+Normally, the identifier is only explicitly created to pass to one the 'Hakyll.Core.Compiler.load' functions.
+
+__Identifiers when using create__.
+Using 'Hakyll.Core.Rules.create' (thereby inventing a file path with no underlying file on disk)
+builds an implicit identifier that corresponds to the invented file path:
+
+@
+create ["index.html"] $ do                            -- saved with implicit identifier 'index.html'
+    makeItem ("Hello world" :: String)
```
Comment (jtojnar) [inline lib/Hakyll/Core/Identifier.hs:73]: Does not typecheck, missing `compile $`?

Comment (jtojnar) [reply lib/Hakyll/Core/Identifier.hs:73]: Same for the examples with `load`, those are also actions in the Compiler monad.

Comment (malteneuss) [reply lib/Hakyll/Core/Identifier.hs]: Thanks. I've correct all locations and hope i haven't missed one

File: lib/Hakyll/Core/Identifier.hs [hunk 2]
```diff
@@ -44,7 +108,7 @@ instance Binary Identifier where
     get = Identifier <$> get <*> get
 
 
---------------------------------------------------------------------------------
+--------------------fromFilePath "posts/hakyll.md"------------------------------------------------------------
```
[the remark this answers is not in the document]
Comment (malteneuss) [reply lib/Hakyll/Core/Identifier.hs]: Yep. Fixed.

File: lib/Hakyll/Core/Identifier.hs [hunk 3]
```diff
@@ -32,6 +35,67 @@ import           GHC.Exts            (IsString, fromString)
...
+@
+-- e.g. file on disk: 'posts\/hakyll.md'
+match "posts/*" $ do                                      -- saved with implicit identifier 'posts\/hakyll.md'
+    compile pandocCompiler
+
+match "about/*" $ do
+    compiledPost <- load (fromFilePath "posts/hakyll.md") -- load with explicit identifier
+    ...
+@
+Normally, the identifier is only explicitly created to pass to one the 'Hakyll.Core.Compiler.load' functions.
```
[the remark this answers is not in the document]
Comment (malteneuss) [reply lib/Hakyll/Core/Identifier.hs]: Fixed

**Review by Minoru — COMMENTED** [review]
Thank you! And thanks to @jtojnar for the review :) I got a couple more comments on top of it, please see below.

**Pull request merged**